modules/calib3d/src/fundam.cpp

```diff
@@ -414,6 +414,11 @@ cv::Mat cv::findHomography( InputArray _points1, InputArray _points2,
 
     if( result && npoints > 4 && method != RHO)
     {
+        // save the original points before compressing
+        const int npoints_input = npoints;
+        const Mat src_input = src.clone();
+        const Mat dst_input = dst.clone();
```

Comment:
Why do you need copies here? HomographyEstimatorCallback::computeError() does not modify inputs.